mlflow/utils/virtualenv.py

```diff
@@ -0,0 +1,254 @@
+    return sorted(matched)[-1]
+
+
+def _install_python(version):
+    """
+    Installs a specified version of python with pyenv and returns a path to the installed python
+    binary.
+
+    :param version: Python version to install.
+    :return: Path to the installed python binary.
+    """
+    version = version if _SEMANTIC_VERSION_REGEX.match(version) else _find_latest_version(version)
+    _logger.info("Installing python %s", version)
+    # pyenv-win doesn't support `--skip-existing` but its behavior is enabled by default
+    # https://github.com/pyenv-win/pyenv-win/pull/314
+    pyenv_install_options = ("--skip-existing",) if _IS_UNIX else ()
+    _exec_cmd(["pyenv", "install", *pyenv_install_options, version], capture_output=False)
+
+    if _IS_UNIX:
+        pyenv_root = _exec_cmd(["pyenv", "root"], capture_output=True).stdout.strip()
+        path_to_bin = ("bin", "python")
+    else:
+        # pyenv-win doesn't provide the `pyenv root` command
+        pyenv_root = os.getenv("PYENV_ROOT")
+        if pyenv_root is None:
+            raise MlflowException("Environment variable 'PYENV_ROOT' must be set")
+        path_to_bin = ("python.exe",)
+    return Path(pyenv_root).joinpath("versions", version, *path_to_bin)
+
+
+def _get_python_env(local_model_path):
+    """
+    Constructs `PythonEnv` from the model artifacts stored in `local_model_path`. If
+    `python_env.yaml` is available, use it, otherwise extract model dependencies from `conda.yaml`.
+    If `conda.yaml` contains conda dependencies except `python`, `pip`, `setuptools`, and, `wheel`,
+    an `MlflowException` is thrown because conda dependencies cannot be installed in a virtualenv
+    environment.
+
+    :param local_model_path: Local directory containing the model artifacts.
+    :return: `PythonEnv` instance.
+    """
+    python_env_file = local_model_path / _PYTHON_ENV_FILE_NAME
+    requirements_file = local_model_path / _REQUIREMENTS_FILE_NAME
+    conda_env_file = local_model_path / _CONDA_ENV_FILE_NAME
+    if python_env_file.exists():
+        return PythonEnv.from_yaml(python_env_file)
+    else:
+        _logger.info(
+            "This model is missing %s, which is because it was logged in an older version"
+            "of MLflow (< 1.26.0) that does not support restoring a model environment with "
+            "virtualenv. Attempting to extract model dependencies from %s and %s instead.",
+            _PYTHON_ENV_FILE_NAME,
+            _REQUIREMENTS_FILE_NAME,
+            _CONDA_ENV_FILE_NAME,
+        )
+        conda_deps = _get_conda_dependencies(conda_env_file)
+        build_packages = ("python", *PythonEnv.BUILD_PACKAGES)
+        conda_deps = [d for d in conda_deps if _get_package_name(d) not in build_packages]
+        if conda_deps:
+            raise MlflowException(
+                f"Cannot restore this model's environment with virtualenv because it contains "
+                f"conda dependencies: {conda_deps}."
+            )
```

Comment:
Left another comment here - https://github.com/mlflow/mlflow/pull/5380#discussion_r844840365. On second thought, I think we should just warn.